docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TimeSeries/DetectSeasonality.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML;
+using Microsoft.ML.TimeSeries;
+
+namespace Samples.Dynamic
+{
+    public static class DetectSeasonality
+    {
+        public static void Example()
+        {
+            /* Create a new ML context, for ML.NET operations. It can be used for
+             exception tracking and logging, as well as the source of randomness.*/
+            var mlContext = new MLContext();
+
+            // Create a seasonal data as input for DetectSeasonality.
+            var dataView = mlContext.Data.LoadFromEnumerable(GetPointsWithSeasonality());
+
+            /* Two option parameters:
+             * seasonalityWindowSize: Default value is -1. When set to -1, use the whole input to fit model; 
+             * when set to a positive integer, only the first windowSize number of values will be considered.
+             * randomnessThreshold: Randomness threashold that specifies how confidence the input values follows 
+             * a predictable pattern recurring over an interval as seasonal data. By default, it is set as 2.81
+             * for 99.5% confidence. The higher the threshold is set, the more strict recurring pattern the 
+             * input values should follow to be determined as seasonal data.
+             */
+            int period = mlContext.AnomalyDetection.DetectSeasonality(
+                dataView,
+                nameof(TimeSeriesData.Value),
+                seasonalityWindowSize: 40);
+
+            // Print the Seasonality Period result.
+            Console.WriteLine($"Seasonality Period: #{period}");
+        }
+
+        private static IEnumerable<TimeSeriesData> GetPointsWithSeasonality()
+        {
```

Comment:
If this feature is designed to detect the period in a set of seasonal time series data, is it possible to illustrate this better with a dataset generated from a periodic function? E.g. `y = sin(2πft)` where the user would know that it is periodic data and can then use this to understand how this feature works?

Reply:
good point, changed to `sin(2 * pi + x)`